js/src/Components/Widgets/LinkWidget.js

```diff
@@ -1,26 +1,27 @@
 import React, {useState} from 'react';
+import {FormHelperText} from "@material-ui/core";
 import TextField from '@material-ui/core/TextField';
 import FormGroup from '@material-ui/core/FormGroup';
 import FormControl from '@material-ui/core/FormControl';
 import FormLabel from '@material-ui/core/FormLabel';
-import {FormHelperText} from "@material-ui/core";
 import Autocomplete from '@material-ui/lab/Autocomplete';
 
 export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) => {
   let timeout;
   let initialCondition = defaultValue;
-  const [urlSuggestions, setSuggestions] = useState([]);
-
-  const emptyLinkValue = {
-    uri: '',
-    title: ''
-  };
 
-  if (initialCondition === undefined || initialCondition.length === 0){
-    initialCondition = [];
-    initialCondition.push(emptyLinkValue);
+  try {
+    // Checking for the length of 0 will catch any null, non-arrays, or empty
+    // arrays. This ensures the initial state is always in a good structure.
+    if (initialCondition.length === 0) {
+      throw 'Default condition failed';
+    }
+  }
+  catch (e) {
+    initialCondition = [{uri: '', title: ''}];
   }
```

Comment:
its just an easy way to skip checking all the different types. and if the `initialCondition` is actually an object instead of an array it helps since `typeof {} == typeof []` is `true` stupid javascript

Reply:
I can do `if (typeof initialCondition === 'object'){...}` since it could be an array or a true object. kinda tricky way to get around all the condition checking.

Reply:
Ah I get you. It just reads funny as you catch your own throw. There is probably a strategy out there for this pattern but I don't know what it is.